explore/src/main/scala/explore/config/SpectroscopyModesTable.scala

```diff
@@ -218,35 +211,77 @@ object SpectroscopyModesTable {
   ): Boolean =
     rowToConf(row).exists(_ === conf)
 
-  protected val tableComponent =
-    ScalaFnComponent[ModesTableProps] { props =>
-      def toggleRow(row: SpectroscopyModeRow): Option[ScienceConfigurationData] =
-        rowToConf(row).filterNot(conf => props.scienceConfiguration.get.contains_(conf))
-
-      val tableInstance = ModesTableMaker.use(props.options)
-      <.div(
-        ExploreStyles.ModesTable,
-        ModesTableComponent(
-          table =
-            Table(celled = true, selectable = true, striped = true, compact = TableCompact.Very)(),
-          header = true,
-          headerCell = (c: ModesTableMaker.ColumnType) =>
-            TableHeaderCell(clazz = ExploreStyles.Sticky |+| ExploreStyles.ModesHeader)(
-              ^.textTransform.capitalize.when(c.id.toString =!= ResolutionColumnId.value),
-              ^.textTransform.none.when(c.id.toString === ResolutionColumnId.value)
-            ),
-          row = (rowData: Row[SpectroscopyModeRow]) =>
-            TableRow(
-              disabled = !enabledRow(rowData.original),
-              clazz = ExploreStyles.ModeSelected.when_(
-                props.scienceConfiguration.get.exists(conf => equalsConf(rowData.original, conf))
+  // protected def ensureContains(matrix: List[SpectroscopyModeRow], conf: Option[ScienceConfigurationData]): List[SpectroscopyModeRow] =
+  //   conf.fold(matrix)(c =>
+  //     if(matrix.exists(row => equalsConf(row, c)))
+  //       matrix
+  //     else
+
+  protected def enabledRow(row: SpectroscopyModeRow): Boolean =
+    List(Instrument.GmosNorth, Instrument.GmosSouth).contains_(row.instrument.instrument) &&
+      row.focalPlane === FocalPlane.SingleSlit
+
+  val component =
+    ScalaFnComponent
+      .withHooks[Props]
+      .useMemoBy(_.spectroscopyRequirements)(props =>
+        s => {
+          val rows                =
+            props.matrix
+              .filtered(
+                focalPlane = s.focalPlane,
+                capabilities = s.capabilities,
+                wavelength = s.wavelength,
+                slitWidth = s.focalPlaneAngle,
+                resolution = s.resolution,
+                range = s.wavelengthRange
+                  .map(_.micrometer.toValue[BigDecimal].toRefined[Positive])
               )
-            )(
-              ^.onClick --> props.scienceConfiguration.set(toggleRow(rowData.original)),
-              props2Attrs(rowData.getRowProps())
-            )
-        )(tableInstance)
+          val (enabled, disabled) = rows.partition(enabledRow)
+          (enabled ++ disabled).toJSArray
+        }
       )
-    }
+      .useMemoBy($ =>
```

Comment:
You can have multiple memo?

Reply:
Yes, you can memoize different values with different dependencies.

Reply:
I thought about using `IO.memoize` for that but didn't follow it through